fiatconnect-api.md

```diff
@@ -383,105 +406,245 @@ The FiatConnect API specifies *two* types of authentication; the first authentic
+the server MUST update the nonce on the session.
 
-* `client`, or *client API key*
+###### 3.2.1.2.3.2. Failure
 
-The semantics of these claims are below:
+On failure, the endpoint MUST return an HTTP `400` error, along with an error code that details exactly why the request failed.
 
-##### 3.2.1.2.1. `"sub"` Claim
+###### 3.2.1.2.3.2.1 `InvalidAddress`
 
-The `sub`, or *subject* claim represents the user's Celo network address. This is a required claim. If it is missing, the server MUST respond
-to the client with an HTTP `400` error. This claim SHOULD be interpreted and used within server endpoints as a unique user identifier, and as the
-source of/destination for crypto funds during transfers. The address should be formatted as 42 hexadecimal characters prefixed by the string literal
-`0x`. If the `sub` claim does not match this format, the server MUST respond to the client with an HTTP `400` error.
+If the provided `address` query parameter is missing or poorly formed, the server MUST return an `InvalidAddress` error.
 
-##### 3.2.1.2.2. `"iss"` Claim
+###### 3.2.1.2.3.2.2 `InvalidDuration`
 
-The `iss`, or *subject* claim represents the user's Celo address public key, dervied from their private key. This is a required claim.
-The public key MUST be in compressed form (see [here](https://docs.ethers.io/v5/api/utils/signing-key/) for more details), and exactly 33 bytes
-long.
+If the provided `sessionDuration` query parameter is missing or poorly formed, the server MUST return an `InvalidAddress` error.
 
-If it is missing, the server MUST respond to the client with an HTTP `400` error. The server MUST use this claim to verify the JWT signature.
-Since the address provided in the `"sub"` claim may not correspond to the public/private keypair used to sign and verify the JWT, the
-server MUST validate that the signature does indeed correspond to the address provided in the `"sub"` claim.
+###### 3.2.1.2.3.2.2 `DurationTooLong`
 
-##### 3.2.1.2.3. `"exp"` Claim
+If the provided `sessionDuration` query parameter represents a duration that is longer than exactly two weeks (1209600 seconds), the server MUST return a `DurationTooLong` error.
+
+#### 3.2.1.3. `POST /auth/login`
+
+The `POST /auth/login` endpoint is responsible for actually verifying signed messages sent by clients, and authenticating the user's session.
+
+##### 3.2.1.3.1. Parameters
+
+###### 3.2.1.3.1.1. Request Body
+
+The *plaintext* request body for this endpoint MUST correspond to a signed SIWE message (as described above) in the [EIP-191](https://eips.ethereum.org/EIPS/eip-191)
+Signed Data Standard format, as required by the SIWE specification.
+
+##### 3.2.1.3.2. Responses
+
+###### 3.2.1.3.2.1. HTTP `200`
+
+On success, the server MUST return an HTTP `200`, and update the state of the user's session to be authenticated.
```

Comment:
The server also needs to return the session cookie, right?

Reply:
Yeah, I'll note that explicitly.

Reply:
Yep, will add this.